SpaceX/app/src/main/java/cz/cvut/popovma1/spacex/feature/rocketlaunch/system/RocketLaunchScreen.kt

```diff
@@ -2,78 +2,126 @@ package cz.cvut.popovma1.spacex.feature.rocketlaunch.system
+        ) {
+/*
+            RocketLaunchAnimation(isLaunched.value)
+            RocketLaunchText(isLaunched.value)
+*/
+            RocketLaunchAnimation(isLifted)
+            RocketLaunchText(isLifted)
+            Log.d("compose: isLifted = $isLifted")
+            Spacer(modifier = Modifier.height(rocketLaunchBottomOffset))
+        }
     }
 }
 
 @Composable
-fun RocketLaunchTopBar(rocketName: String, onBackClick: () -> Unit) {
-    CenteredTitleTopBar(title = stringResource(id = R.string.rocket_launch_title)) {
-        BackButton(text = rocketName, onBackClick = onBackClick)
-    }
+private fun RocketLaunchAnimation(isLaunched: Boolean) {
+    val flyOffset: Dp by animateDpAsState(
+        targetValue = if (isLaunched) rocketFlyingOffset else rocketIdleOffset,
+        animationSpec = tween(rocketLaunchDuration)
+    )
+    Image(
+        painter = getLaunchPainter(isLaunched),
+        contentDescription = stringResource(id = R.string.rocket_launch_image_description),
+        modifier = Modifier
+            .width(launchedRocketWidth)
+            .height(launchedRocketMaxHeight)
+            .offset(y = flyOffset),
+        contentScale = ContentScale.FillWidth,
+        alignment = Alignment.TopCenter
+    )
 }
 
 @Composable
-fun RocketLaunchContent() {
-    Column(
-        modifier = Modifier.fillMaxSize(),
-        horizontalAlignment = Alignment.CenterHorizontally,
-    ) {
-        Column(
-            modifier = Modifier.weight(1f),
-            verticalArrangement = Arrangement.Bottom
-        ) {
-            Image(
-                painter = painterResource(id = R.drawable.ic_rocket_idle),
-                contentDescription = stringResource(id = R.string.rocket_launch_image_description),
-                modifier = Modifier
-                    .width(launchedRocketSize)
-//                    .padding(bottom = 1.dp) // TODO use in launch animation ?
-            )
-//            Spacer(modifier = Modifier.height(0.dp)) // TODO use in launch animation ?
-        }
-        Text(
-            text = stringResource(id = getLaunchText(false)),
-            modifier = Modifier.width(launchedRocketSize)
-        )
-        Spacer(modifier = Modifier.height(200.dp))
-    }
+private fun RocketLaunchText(isLaunched: Boolean) {
+    Text(
+        modifier = Modifier.size(
+            width = rocketLaunchWidth,
+            height = rocketLaunchHeight
+        ),
+        text = getLaunchText(isLaunched),
+        textAlign = TextAlign.Center
+    )
 }
 
-private fun getLaunchText(isLaunched: Boolean): Int =
-    if(isLaunched)
-        R.string.rocket_launch_text_before_launch
+@Composable
+private fun getLaunchPainter(isLaunched: Boolean) =
+    if (isLaunched)
+        painterResource(id = R.drawable.ic_rocket_flying)
     else
-        R.string.rocket_launch_text_after_launch
+        painterResource(id = R.drawable.ic_rocket_idle)
 
+@Composable
+private fun getLaunchText(isLaunched: Boolean) =
```

Comment:
Přijde mi to až zbytečně moc rozkouskované. obsah téhle funkce bych nechala v RocketLaunchText.